extra/helpers.py

```diff
@@ -28,10 +30,17 @@ def proc(itermaker, q) -> None:
   q.put(None)
   q.close()
 
+class _CloudpickleFunctionWrapper:
+  def __init__(self, fn):
+    self.pfn = cloudpickle.dumps(fn)
+
+  def __call__(self) -> Any:
+    return cloudpickle.loads(self.pfn)()
```

Comment:
This will unpick the function each time!

Reply:
Oops 😅 moved pickling to `__getstate__` and `__setstate__`.